e2e/windows/e2e-starter.sh

```diff
@@ -23,7 +23,7 @@ if [ -n "$out" ]; then
     provisioning_state=$(az aks show -n $CLUSTER_NAME -g $RESOURCE_GROUP_NAME -ojson | jq '.provisioningState' | tr -d "\"")
     MC_RG_NAME="MC_${RESOURCE_GROUP_NAME}_${CLUSTER_NAME}_$LOCATION"
     exists=$(az group exists -n $MC_RG_NAME)
-    if [ "$exists" == "false" ] || [ "$provisioning_state" == "Failed" ]; then
+    if [ "$exists" == "false" ] || [ "$provisioning_state" == "Failed" ] || [ "$provisioning_state" == "Canceled" ]; then
```

Comment:
https://learn.microsoft.com/en-us/rest/api/aks/managed-clusters/get?tabs=HTTP#managedclusterpodidentityprovisioningstate
Will the pipeline wait if `$provisioning_state` is `Deleting`, `Updating`, etc.?

Reply:
Thanks for the comment. Add the case of deleting and updating too.